airbyte-integrations/connectors/source-google-ads/source_google_ads/utils.py

```diff
@@ -41,6 +50,83 @@
 }
 
 
+class ExpiredPageTokenError(AirbyteTracedException):
+    """
+    Custom AirbyteTracedException exception to handle the scenario when the page token has expired
+    while processing a response from Google Ads.
+    """
+
+    pass
+
+
+def is_error_type(error_value, target_enum_value):
+    """Compares error value with target enum value after converting both to integers."""
+    return int(error_value) == int(target_enum_value)
+
+
+def traced_exception(ga_exception: GoogleAdsException, customer_id: str, catch_disabled_customer_error: bool):
+    """Add user-friendly message for GoogleAdsException"""
+    unknown_error = False
+    for error in ga_exception.failure.errors:
```

Comment:
This algorithm seems a bit weird to me. My understanding is that we will raise on the first error code we are handling ignoring all the others. I'm wondering if it'll provide good UX. For example, if error codes are `USER_PERMISSION_DENIED` and `CUSTOMER_NOT_ENABLED`, should we tell the user both? Else, we will probably first the first one, retry a sync and it'll still fail for an error we knew about

Reply:
This approach aimed to avoid overwhelming the user with multiple errors at once, allowing them to address one issue at a time. Some errors, such as query errors, can be lengthy. Presenting one error at a time would provide a clearer and more manageable user experience. Should I merge them?

Reply:
Than I think it's a balance of "not overwhelming the user" and "him not having to run multiple syncs to fix all the issues". I don't know where this balance lies as I don't know much about the API but it feels like having to re-run a sync to know the information is increasing way more the feedback loop than having to read through a big error message. I'll let you be the judge of that but if you agree with the previous statement, I would probably merge the error messages

Reply:
Add merging errors